### ISSUE TITLE
number filter on '0' (equals/not-equals) not working

### ISSUE THREAD

**Issue by kvloover** [ag-grid/ag-grid#4863]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```
**Providing a Reproducible Scenario**
https://plnkr.co/edit/aWNPm8Dwjb7zHAfQ?preview

Added a simple row with 0 sale to the number filter example.

**Current behavior**
equals and not-equals using '0' as input does not filter properly.
- Equals: removes all lines including the 0.
- Not-equals: does not remove the 0 line.

**Expected behavior**
- Equals to show only the 0 line
- Not-equals to remove the 0 line

**AG Grid version:** 26.2.0

**Browser:**
Microsoft-Edge:
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.110 Safari/537.36 Edg/96.0.1054.62

**Comment by yusufkayaer** [comment]
Hi,
AG-6052
[Regression] When using an agNumberColumnFilter to filter for values equal to 0, no values are returned 
https://www.ag-grid.com/pipeline/

**Comment by makinggoodsoftware** [comment]
Hi,

In order to help us keep our issues queue clear, we are going to close this issue

The official way to check the status for an issue raised here is to check its AG-xxxx ID against our pipeline and changelog

…the pipeline (if not yet developed)

https://www.ag-grid.com/ag-grid-pipeline/

... or the changelog, (if the issue has been resolved)

https://www.ag-grid.com/ag-grid-changelog/

We keep issues raised in the pipeline through github open for a while so that the community can add some feedback to them, and close them when there is no activity for a while.

If you want to add further feedback to a closed ticket with this comment, we would recommend opening a new ticket as we don’t monitor closed issues.

**Comment by alikleit** [comment]
I cannot se the `AG-6052` in the pipline anymore, and the issue seems to be closed @makinggoodsoftware 

but, This is still happening in `33.2.4`  ... 

Any take on this? 

Thanks in advance.

**Comment by AG-Zoheil** [comment]
Hi

Thank you for your comment.

We'd be keen to look into this again, but as you can see this is quite an old issue from **more than three years ago.**

AG-6052 was closed in version 27.3.0.

The grid has naturally advanced a lot since then. If you are still having problems, can we please ask you to open a new issue with a reproducible live scenario - you can use one of the plunker examples from our [documentation](https://ag-grid.com/javascript-data-grid/) as a starting point.

This would help our team support in the most efficient way, and also increase visibility for the wider AG Grid community in case of contributions and comments.

Much appreciated!

Kind regards,
Zoheil
